src/celeritas/ext/detail/GenericSolid.hh

```diff
@@ -16,12 +16,22 @@
 #include <VecGeom/volumes/PlacedVolume.h>
 #include <VecGeom/volumes/UnplacedVolume.h>
 
+#include "corecel/Assert.hh"
+
 #include "GenericPlacedVolume.hh"
 
 namespace celeritas
 {
 namespace detail
 {
+//---------------------------------------------------------------------------//
+/*!
+ * Base class for identifying "generic" VecGeom volumes.
+ */
+class GenericSolidBase : public vecgeom::VUnplacedVolume
+{
+};
```

Comment:
Why is this extra inheritance necessary if base class is empty?

Reply:
It's used to `dynamic_cast` when choosing not to print the "incompatible volume" warning: having a common base class for all the templated daughters seemed like the easiest solution to me, but I'm open to other suggestions.